server/routerlicious/packages/services-telemetry/src/lumberjack.ts

```diff
@@ -0,0 +1,111 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import path from "path";
+import { LumberEventName } from "./lumberEventNames";
+import { Lumber } from "./lumber";
+import { LogLevel, LumberType, ITelemetryMetadata, ILumberjackEngine } from "./resources";
+
+// Lumberjack is a telemetry manager class that allows the collection of metrics and logs
+// throughout the service. A list of ILumberjackEngine must be provided to Lumberjack
+// by calling setupEngines() before Lumberjack can be used.
+export class Lumberjack {
+    protected static _instance: Lumberjack | undefined = undefined;
+    private readonly engineList: ILumberjackEngine[];
+    private isSetupCompleted: boolean;
+
+    protected constructor() {
+        this.engineList = [];
+        this.isSetupCompleted = false;
+    }
+
+    public static get instance() {
```

Comment:
isn't it possible users of this class may want more than 1 instance of the lumberjack?

Reply:
Lumberjack is supposed to be globally available and provide the same list of engines throughout the Node.js process. To accomplish that, we could either make it a singleton or a static class, and we decided to use a singleton. So we do not want more than 1 instance of Lumberjack to be available for the given Node.js process.

Reply:
Do we have to restrict the framework's use to one set of engines and 1 global instance? There may be scenarios where it could be handy to split multiple instances with different sets of engines (eg. different background scheduled jobs, workers).

Reply:
I think Tri makes a good point. Other libraries that expose a global instance also allow for the manual creation of other instances. The user is then responsible for how that other instance is passed around and accessed. For example, [Axios](https://github.com/axios/axios#creating-an-instance) and [Winston](https://github.com/winstonjs/winston#creating-your-own-logger). If it's not too difficult to expose a way for a new LumberJack to be created, while also exposing a default global instance, it seems like a good idea

Reply:
There should be 1 Lumberjack instance per Node.js process. When we think about microservices, that means that each set: Scribe, Scriptorium, Deli, Alfred, etc. would be able to define their own Lumberjack setup. But it is true that, if possible, it would be nice to allow for a secondary Lumberjack instance at some point within each process. Let me research how exactly to do that.

Reply:
Similarly to Axios' `create` and Winston's `createLogger`, which allow the creation of individual instances, I added a `create` static method in Lumberjack which takes a list of engines and returns a new, separate Lumberjack instance. I also added some UTs. Let me know what you think or if it should be done in a different way!

Reply:
Love this idea. Thanks @TriDucNgo for suggesting it